### PR TITLE
Add support for Ruby 3.0.0

### PR DESCRIPTION
**Pull request by pblesi**
This addresses issue: https://github.com/braintree/runbook/issues/40

In addition to adding support for Ruby 3.0.0, this commit does the
following:

* Adds support for Ruby 2.7.2
* Removes support for Ruby 2.2.10
* Adds support for Bundler < 3.0
* Drops support for Bundler < 2.2.0
* Adds a ci option to the project generator

### CODE REVIEW

**Comment by coopergillan** [comment]
@pblesi - this one now needs a merge conflict change due to gem upgrades.

**Review by scannillo — APPROVED** [review]
👋  Thanks for making this update. The Braintree SDK team uses Runbook (via gist) to release our Ruby SDK, which recently bumps to support 2.6+. So this PR should also fix a current issue with our release script.

If you need any additional 👀  or help on it, let us know!

**Review by ClashTheBunny — APPROVED** [review]
:shipit:

File: .travis.yml [hunk 1]
```diff
@@ -3,26 +3,23 @@ language: ruby
 services:
   - docker
 rvm:
-  - 2.2.10
   - 2.3.8
-  - 2.4.6
-  - 2.5.5
-  - 2.6.3
+  - 2.4.10
```
Comment (hollabaq86) [inline .travis.yml:7]: the Ruby core community [dropped support](https://www.ruby-lang.org/en/downloads/branches/) for versions below 2.6, I'm thinking we should drop any versions below 2.6, as well.

File: runbook.gemspec [hunk 2]
```diff
@@ -42,7 +43,7 @@ Gem::Specification.new do |spec|
 
   spec.add_development_dependency "appraisal", "~> 2.2"
   spec.add_development_dependency "aruba", "~> 0.14"
-  spec.add_development_dependency "bundler", "~> 1.15"
+  spec.add_development_dependency "bundler", ">= 2.2.0", "< 3.0"
```
Comment (hollabaq86) [inline runbook.gemspec:46]: why not use `~> 2.2` here, instead?

**Review by hollabaq86 — COMMENTED** [review]
@pblesi ❤️  this work updating runbook. I left a few questions/comments around versioned testing and dependencies.

File: .travis.yml (hunk 1 above)
Comment (pblesi) [reply .travis.yml:7]: I would prefer to continue to support these versions as long as it is not inconvenient. On the off chance that someone needs a feature from this version, but they find it difficult to upgrade their Ruby version, I want to be accommodating to their needs.

File: runbook.gemspec (hunk 2 above)
Comment (pblesi) [reply runbook.gemspec:46]: Yeah, that makes sense to me. I can't recall why I expressed it this way. Will update.

File: .travis.yml (hunk 1 above)
Comment (ClashTheBunny) [reply .travis.yml:7]: Debian oldstable, still in lts, is ruby 2.3.3.

Comment (hollabaq86) [reply .travis.yml:7]: _wowwwwwwwww_

**Review by hollabaq86 — APPROVED** [review]

**Pull request merged**